plugins/core/jetbrains-community/src/software/aws/toolkits/jetbrains/services/telemetry/otel/OtelBase.kt

```diff
@@ -0,0 +1,243 @@
+    ): T =
+        ijUseWithScope(context) { span ->
+            operation(span as SpanType)
+        }
+
+    protected var parent: Context? = null
+    override fun setParent(context: Context): BuilderType {
+        parent = context
+        delegate.setParent(context)
+        return this as BuilderType
+    }
+
+    override fun setNoParent(): BuilderType {
+        parent = null
+        delegate.setNoParent()
+        return this as BuilderType
+    }
+
+    override fun addLink(spanContext: SpanContext): BuilderType {
+        delegate.addLink(spanContext)
+        return this as BuilderType
+    }
+
+    override fun addLink(
+        spanContext: SpanContext,
+        attributes: Attributes,
+    ): BuilderType {
+        delegate.addLink(spanContext, attributes)
+        return this as BuilderType
+    }
+
+    override fun setAttribute(key: String, value: String): BuilderType {
+        delegate.setAttribute(key, value)
+        return this as BuilderType
+    }
+
+    override fun setAttribute(key: String, value: Long): BuilderType {
+        delegate.setAttribute(key, value)
+        return this as BuilderType
+    }
+
+    override fun setAttribute(key: String, value: Double): BuilderType {
+        delegate.setAttribute(key, value)
+        return this as BuilderType
+    }
+
+    override fun setAttribute(key: String, value: Boolean): BuilderType {
+        delegate.setAttribute(key, value)
+        return this as BuilderType
+    }
+
+    override fun <V : Any?> setAttribute(
+        key: AttributeKey<V?>,
+        value: V & Any,
+    ): BuilderType {
+        delegate.setAttribute(key, value)
+        return this as BuilderType
+    }
+
+    override fun setAllAttributes(attributes: Attributes): BuilderType {
+        delegate.setAllAttributes(attributes)
+        return this as BuilderType
+    }
+
+    override fun setSpanKind(spanKind: SpanKind): BuilderType {
+        delegate.setSpanKind(spanKind)
+        return this as BuilderType
+    }
+
+    override fun setStartTimestamp(startTimestamp: Long, unit: TimeUnit): BuilderType {
+        delegate.setStartTimestamp(startTimestamp, unit)
+        return this as BuilderType
+    }
+
+    override fun setStartTimestamp(startTimestamp: Instant): BuilderType {
+        delegate.setStartTimestamp(startTimestamp)
+        return this as BuilderType
+    }
+
+    protected abstract fun doStartSpan(): SpanType
+
+    override fun startSpan(): SpanType {
+        var parent = parent
+        if (parent == null) {
+            parent = Context.current()
+        }
+        requireNotNull(parent)
+
+        val contextValue = parent.get(AWS_PRODUCT_CONTEXT_KEY)
+        if (contextValue == null) {
+            val s = Span.fromContextOrNull(parent)
+            parent = if (s is AbstractBaseSpan<*> && s.context != null) {
+                s.context.with(Span.fromContext(parent))
+            } else {
+                parent.with(AWS_PRODUCT_CONTEXT_KEY, resolvePluginName())
```

Comment:
Should resolvePluginName be under a try catch?

Reply:
returns null if fails